backend/api/community_create.go

```diff
@@ -0,0 +1,113 @@
+package api
+
+import (
+	"database/sql"
+	"errors"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	db "github.com/git-adithyanair/cs130-group-project/db/sqlc"
+	"github.com/git-adithyanair/cs130-group-project/token"
+)
+
+type CreateCommunityRequestStore struct {
+	Name    string  `json:"name" binding:"required"`
+	XCoord  float64 `json:"x_coord" binding:"required,numeric"`
+	YCoord  float64 `json:"y_coord" binding:"required,numeric"`
+	PlaceID string  `json:"place_id" binding:"required"`
+	Address string  `json:"address" binding:"required"`
+}
+
+type CreateCommunityRequest struct {
+	Name         string                        `json:"name" binding:"required"`
+	CenterXCoord float64                       `json:"center_x_coord" binding:"required,numeric"`
+	CenterYCoord float64                       `json:"center_y_coord" binding:"required,numeric"`
+	Range        int32                         `json:"range" binding:"required,numeric"`
+	PlaceID      string                        `json:"place_id" binding:"required"`
+	Address      string                        `json:"address" binding:"required"`
+	Stores       []CreateCommunityRequestStore `json:"stores" binding:"required"`
+}
+
+func (server *Server) CreateCommunity(ctx *gin.Context) {
+	var req CreateCommunityRequest
+	if err := ctx.ShouldBindJSON(&req); err != nil {
+		ctx.JSON(http.StatusBadRequest, errorResponse(err))
+		return
+	}
+
+	_, err := server.queries.GetCommunityByPlaceID(ctx, req.PlaceID)
+	if err != nil {
+		if err == sql.ErrNoRows {
+
+			authPayload := ctx.MustGet(authorizationPayloadKey).(*token.Payload)
+
+			arg := db.CreateCommunityParams{
+				Name:         req.Name,
+				Admin:        authPayload.UserID,
+				CenterXCoord: req.CenterXCoord,
+				CenterYCoord: req.CenterYCoord,
+				Range:        req.Range,
+				PlaceID:      req.PlaceID,
+				Address:      req.Address,
+			}
+
+			community, err := server.queries.CreateCommunity(ctx, arg)
+			if err != nil {
+				if err == sql.ErrNoRows {
```

Comment:
Is this error possible when creating a row?

Reply:
great point lmaooo